main/src/library/Foldable.flix

```diff
@@ -80,24 +80,24 @@ pub class Foldable[t : Type -> Type] {
     ///
     /// Returns `false` if `t` is empty.
     ///
-    pub def exists(f: a -> Bool, t: t[a]): Bool =
-        Foldable.foldLeft((acc, x) -> acc or f(x), false, t)
+    pub def exists(f: a -> Bool, t: t[a]): Bool & ef =
+        Foldable.foldRightWithCont((x, k) -> k() or f(x), false, t)
```

Comment:
should the effect of the functions be `ef` or should `foldRightWithCont` be cast as `Pure`?

Reply:
All functions should now have an effect `ef`.